controllers/topology/desired_state.go

```diff
@@ -95,6 +96,8 @@ func computeInfrastructureCluster(_ context.Context, s *scope.Scope) (*unstructu
 		cluster:               cluster,
 		namePrefix:            fmt.Sprintf("%s-", cluster.Name),
 		currentObjectRef:      currentRef,
+		// Note: It is not possible to add an ownerRef to Cluster at this stage, otherwise the provisioning
```

Comment:
is this referring to e.g https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/26589e22fd6b57d0946a14f6535069858ef3ed9b/controllers/awscluster_controller.go#L99-L102
Do we have a doc we can point at where this is documented as a contract?

Reply:
this is documented here https://cluster-api.sigs.k8s.io/developer/providers/implementers-guide/controllers_and_reconciliation.html#reconciliation 
eventual doc can be improved documenting the same point also in https://cluster-api.sigs.k8s.io/developer/architecture/controllers/cluster.html, TBh I don't consider this in scope of this PR.

Reply:
Sure! I didn't mean to change docs in this PR but I was just wondering if there was something we could link in the code comment. Otherwise I suspect new joiners with few context might have a hard time to follow. I usually imagine people reading my code / code comments / commits with zero context so the more obvious the easier for others to get involved. I guess a link to https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html#normal-resource (2 step) or the one to code above might help. Just a minor thought no need to change here :)
fwiw I also created https://github.com/kubernetes-sigs/cluster-api/pull/5611